diff --git a/src/disp/dftd4.f90 b/src/disp/dftd4.f90
index ec043ac..f5203c9 100644
--- a/src/disp/dftd4.f90
+++ b/src/disp/dftd4.f90
@@ -924,8 +924,8 @@ subroutine weight_references(dispm, nat, atoms, g_a, g_c, wf, q, cn, &
    zerodcn = 0.0_wp
    zetadq  = 0.0_wp
 
-   !$omp parallel do default(none) shared(zetavec, zetadcn, zetadq, zerodcn) &
-   !$omp shared(nat, atoms, zeff, gam, dispm, cn, q) &
+   !$omp parallel do default(shared) shared(zetavec, zetadcn, zetadq, zerodcn) &
+   !$omp shared(nat, atoms, dispm, cn, q) &
    !$omp private(iat, ati, zi, gi, norm, dnorm, iref, icount, twf, gw, expw, &
    !$omp& expd, gwk, dgwk)
    do iat = 1, nat
@@ -1013,7 +1013,7 @@ subroutine get_atomic_c6(dispm, nat, atoms, zetavec, zetadcn, zetadq, &
    !$omp parallel do default(none) shared(c6, dc6dcn, dc6dq) &
    !$omp shared(nat, atoms, dispm, zetavec, zetadcn, zetadq) &
    !$omp private(iat, ati, jat, atj, dc6, dc6dcni, dc6dcnj, dc6dqi, dc6dqj, &
-   !$omp& iref, jref)
+   !$omp& iref, jref, refc6)
    do iat = 1, nat
       ati = atoms(iat)
       do jat = 1, iat
@@ -1461,7 +1461,7 @@ subroutine atm_gradient_neigh &
    real(wp), parameter :: sr = 4.0_wp/3.0_wp
 
    !$omp parallel do default(none) reduction(+:energies, gradient, sigma, dEdcn) &
-   !$omp shared(mol, neighs, neighlist, par, r4r2) &
+   !$omp shared(mol, neighs, neighlist, par, r4r2, c6, dc6dcn) &
    !$omp private(iat, ati, ij, jtr, r2ij, rij, jat, atj, ik, ktr, kat, atk, rik, &
    !$omp& r2ik, rjk, r2jk, c6ij, cij, c6ik, c6jk, cik, cjk, scale, dE, dG, dS, dCN)
    do iat = 1, len(mol)
@@ -1932,10 +1932,10 @@ subroutine atm_gradient_latp &
 
    cutoff2 = cutoff**2
 
-   !$omp parallel do default(none) reduction(+energies, gradient, sigma, dEdcn) &
-   !$omp shared(mol, r4r2, par, trans, cutoff2) &
+   !$omp parallel do default(none) reduction(+:energies, gradient, sigma, dEdcn) &
+   !$omp shared(mol, r4r2, par, trans, cutoff2, c6, dc6dcn) &
    !$omp private(iat, ati, jat, atj, kat, atk, c6ij, cij, c6ik, c6jk, cik, cjk, &
-   !$omp& itr, rij, r2ij, ktr, rik, r2ik, rjk, r2jk, scale, dE, dG, dS, dCN)
+   !$omp& rij, r2ij, ktr, rik, r2ik, rjk, r2jk, scale, dE, dG, dS, dCN)
    do iat = 1, len(mol)
       ati = mol%at(iat)
       do jat = 1, iat
