diff --git a/TESTSUITE/meson.build b/TESTSUITE/meson.build
index 0f857ab..1551d08 100644
--- a/TESTSUITE/meson.build
+++ b/TESTSUITE/meson.build
@@ -59,7 +59,7 @@ test('Info', xtb_exe, env: xtbenv, args: [
   files('../assets/inputs/xyz/taxol.xyz'),
 ])
 
-test('Singlepoint', xtb_exe, args: ['--coffee', '--strict', '--norestart', '--namespace', 'test1'], env: xtbenv)
+test('Singlepoint', xtb_exe, args: ['--coffee', '--strict', '--norestart', '--namespace', 'test1', '--gfn', '2'], env: xtbenv)
 if get_option('la_backend') != 'netlib'
   test('Geometry opt.', xtb_exe, args: ['--coffee', '--opt', '--strict', '--norestart', '--namespace', 'test2'], env: xtbenv)
 endif
