diff --git a/src/filetools.f90 b/src/filetools.f90
index 69cbf3a..b9b7f30 100644
--- a/src/filetools.f90
+++ b/src/filetools.f90
@@ -207,12 +207,13 @@ subroutine open_file(iunit,name,status)
 end subroutine open_file
 
 subroutine close_file(unit)
+   use xtb_mctc_io
    use xtb_mctc_filetools
    implicit none
    integer,intent(in) :: unit
    logical :: opened
    !$omp critical (io)
-   inquire(unit=unit,opened=opened)
+   opened = any(unit /= [-1, stderr, stdout, stdin])
    if (opened) then
       close(unit)
       call pop_file(unit)
@@ -221,12 +222,13 @@ subroutine close_file(unit)
 end subroutine close_file
 
 subroutine remove_file(unit)
+   use xtb_mctc_io
    use xtb_mctc_filetools
    implicit none
    integer,intent(in) :: unit
    logical :: opened
    !$omp critical (io)
-   inquire(unit=unit,opened=opened)
+   opened = any(unit /= [-1, stderr, stdout, stdin])
    if (opened) then
       close(unit,status='delete')
       call pop_file(unit,'d')
