From 5eb967ddc76b7e10112fa2bb83e294b0f4968609 Mon Sep 17 00:00:00 2001
From: "Igor S. Gerasimov" <foxtranigor@gmail.com>
Date: Mon, 12 May 2025 01:56:27 +0200
Subject: [PATCH] Fix commas in optimizer

Signed-off-by: Igor S. Gerasimov <foxtranigor@gmail.com>
---
diff --git a/src/optimizer.f90 b/src/optimizer.f90
index e6fdc9a..7b49547 100644
--- a/src/optimizer.f90
+++ b/src/optimizer.f90
@@ -849,7 +849,7 @@ subroutine relax(env,iter,mol,anc,restart,maxcycle,maxdispl,ethr,gthr, &
       write(env%unit,'(5x,"change   ",e18.7,1x,"Eh")')                      echng
       write(env%unit,'(3x,"gradient norm :",f14.7,1x,"Eh/α")',advance='no') gnorm
       write(env%unit,'(3x,"predicted",e18.7)',advance='no')                 depred
-      write(env%unit,'(1x,"("f7.2"%)")')         (depred-echng)/echng*100
+      write(env%unit,'(1x,"(",f7.2,"%)")')       (depred-echng)/(echng+1e-34_wp)*100
    endif
    
    ! check 0 energy case !
